### PR TITLE
cxx-qt-lib: qdatetime put deprecated behind cfg flag

### LINKED ISSUE #1106: Qt 6.8 warnings due to deprecations cause build failure
When building with Qt 6.8.0 the following warnings occur due to deprecations in those methods.

<details>

<summary>Qt 6.8 warnings</summary>

```
warning: cxx-qt-lib@0.6.1: /home/runner/work/cxx-qt/cxx-qt/build/./cargo/build/x86_64-unknown-linux-gnu/release/build/cxx-qt-lib-3511ff13ff3509ed/out/cxx-qt-gen/src/core/qdatetime.cxx.cpp: In function ‘void cxxbridge1$QDateTime$set_offset_from_utc(QDateTime&, int32_t)’:
warning: cxx-qt-lib@0.6.1: /home/runner/work/cxx-qt/cxx-qt/build/./cargo/build/x86_64-unknown-linux-gnu/release/build/cxx-qt-lib-3511ff13ff3509ed/out/cxx-qt-gen/src/core/qdatetime.cxx.cpp:110:77: warning: ‘void QDateTime::setOffsetFromUtc(int)’ is deprecated: Use setTimeZone() instead [-Wdeprecated-declarations]
warning: cxx-qt-lib@0.6.1:   110 |   void (::QDateTime::*set_offset_from_utc$)(::std::int32_t) = &::QDateTime::setOffsetFromUtc;
warning: cxx-qt-lib@0.6.1:       |                                                                             ^~~~~~~~~~~~~~~~
warning: cxx-qt-lib@0.6.1: In file included from /home/runner/work/cxx-qt/Qt/6.8.0/gcc_64/include/QtCore/QDate:1,
warning: cxx-qt-lib@0.6.1:                  from /home/runner/work/cxx-qt/cxx-qt/build/./cargo/build/x86_64-unknown-linux-gnu/release/build/cxx-qt-lib-3511ff13ff3509ed/out/cxx-qt-build/target/crates/cxx-qt-lib/include/cxx-qt-lib/qdate.h:12,
warning: cxx-qt-lib@0.6.1:                  from /home/runner/work/cxx-qt/cxx-qt/build/./cargo/build/x86_64-unknown-linux-gnu/release/build/cxx-qt-lib-3511ff13ff3509ed/out/cxx-qt-gen/src/core/qdatetime.cxx.cpp:2:
warning: cxx-qt-lib@0.6.1: /home/runner/work/cxx-qt/Qt/6.8.0/gcc_64/include/QtCore/qdatetime.h:406:10: note: declared here
warning: cxx-qt-lib@0.6.1:   406 |     void setOffsetFromUtc(int offsetSeconds);
warning: cxx-qt-lib@0.6.1:       |          ^~~~~~~~~~~~~~~~
warning: cxx-qt-lib@0.6.1: /home/runner/work/cxx-qt/cxx-qt/build/./cargo/build/x86_64-unknown-linux-gnu/release/build/cxx-qt-lib-3511ff13ff3509ed/out/cxx-qt-gen/src/core/qdatetime.cxx.cpp: In function ‘void cxxbridge1$QDateTime$set_time_spec(QDateTime&, Qt::TimeSpec*)’:
warning: cxx-qt-lib@0.6.1: /home/runner/work/cxx-qt/cxx-qt/build/./cargo/build/x86_64-unknown-linux-gnu/release/build/cxx-qt-lib-3511ff13ff3509ed/out/cxx-qt-gen/src/core/qdatetime.cxx.cpp:115:71: warning: ‘void QDateTime::setTimeSpec(Qt::TimeSpec)’ is deprecated: Use setTimeZone() instead [-Wdeprecated-declarations]
warning: cxx-qt-lib@0.6.1:   115 |   void (::QDateTime::*set_time_spec$)(::Qt::TimeSpec) = &::QDateTime::setTimeSpec;
warning: cxx-qt-lib@0.6.1:       |                                                                       ^~~~~~~~~~~
warning: cxx-qt-lib@0.6.1: /home/runner/work/cxx-qt/Qt/6.8.0/gcc_64/include/QtCore/qdatetime.h:404:10: note: declared here
warning: cxx-qt-lib@0.6.1:   404 |     void setTimeSpec(Qt::TimeSpec spec);
warning: cxx-qt-lib@0.6.1:       |          ^~~~~~~~~~~
warning: cxx-qt-lib@0.6.1: /home/runner/work/cxx-qt/cxx-qt/build/./cargo/build/x86_64-unknown-linux-gnu/release/build/cxx-qt-lib-3511ff13ff3509ed/out/cxx-qt-gen/src/core/qdatetime.cxx.cpp: In function ‘void cxxbridge1$QDateTime$to_time_spec(const QDateTime&, Qt::TimeSpec*, QDateTime*)’:
warning: cxx-qt-lib@0.6.1: /home/runner/work/cxx-qt/cxx-qt/build/./cargo/build/x86_64-unknown-linux-gnu/release/build/cxx-qt-lib-3511ff13ff3509ed/out/cxx-qt-gen/src/core/qdatetime.cxx.cpp:145:83: warning: ‘QDateTime QDateTime::toTimeSpec(Qt::TimeSpec) const’ is deprecated: Use toTimeZone instead [-Wdeprecated-declarations]
warning: cxx-qt-lib@0.6.1:   145 |   ::QDateTime (::QDateTime::*to_time_spec$)(::Qt::TimeSpec) const = &::QDateTime::toTimeSpec;
warning: cxx-qt-lib@0.6.1:       |                                                                                   ^~~~~~~~~~
warning: cxx-qt-lib@0.6.1: /home/runner/work/cxx-qt/Qt/6.8.0/gcc_64/include/QtCore/qdatetime.h:436:15: note: declared here
warning: cxx-qt-lib@0.6.1:   436 |     QDateTime toTimeSpec(Qt::TimeSpec spec) const;
warning: cxx-qt-lib@0.6.1:       |               ^~~~~~~~~~
warning: cxx-qt-lib@0.6.1: In file included from /home/runner/work/cxx-qt/cxx-qt/build/./cargo/build/x86_64-unknown-linux-gnu/release/build/cxx-qt-lib-3511ff13ff3509ed/out/cxx-qt-gen/src/core/qdatetime.cxx.cpp:7:
warning: cxx-qt-lib@0.6.1: /home/runner/work/cxx-qt/cxx-qt/build/./cargo/build/x86_64-unknown-linux-gnu/release/build/cxx-qt-lib-3511ff13ff3509ed/out/cxx-qt-build/target/crates/cxx-qt-lib/include/cxx-qt-lib/common.h: In instantiation of ‘T rust::cxxqtlib1::construct(Args ...) [with T = QDateTime; Args = {const QDate&, const QTime&, Qt::TimeSpec, int}]’:
warning: cxx-qt-lib@0.6.1: /home/runner/work/cxx-qt/cxx-qt/build/./cargo/build/x86_64-unknown-linux-gnu/release/build/cxx-qt-lib-3511ff13ff3509ed/out/cxx-qt-gen/src/core/qdatetime.cxx.cpp:279:150:   required from here
warning: cxx-qt-lib@0.6.1: /home/runner/work/cxx-qt/cxx-qt/build/./cargo/build/x86_64-unknown-linux-gnu/release/build/cxx-qt-lib-3511ff13ff3509ed/out/cxx-qt-build/target/crates/cxx-qt-lib/include/cxx-qt-lib/common.h:23:10: warning: ‘QDateTime::QDateTime(QDate, QTime, Qt::TimeSpec, int)’ is deprecated: Pass QTimeZone instead [-Wdeprecated-declarations]
warning: cxx-qt-lib@0.6.1:    23 |   return T(args...);
warning: cxx-qt-lib@0.6.1:       |          ^~~~~~~~~~
warning: cxx-qt-lib@0.6.1: /home/runner/work/cxx-qt/Qt/6.8.0/gcc_64/include/QtCore/qdatetime.h:359:5: note: declared here
warning: cxx-qt-lib@0.6.1:   359 |     QDateTime(QDate date, QTime time, Qt::TimeSpec spec, int offsetSeconds = 0);
warning: cxx-qt-lib@0.6.1:       |     ^~~~~~~~~
```

</details>

### PR DESCRIPTION
**Pull request by ahayzen-kdab**
Closes #1106

### CODE REVIEW

File: crates/cxx-qt-lib/src/core/qdatetime.rs
Comment (ahayzen-kdab) [inline crates/cxx-qt-lib/src/core/qdatetime.rs:56]: remaining issue appears to be with Qt 6.8 the linking order is incorrect with Qt6QuickControls2 being linked before our staticlib

File: crates/cxx-qt-build/src/lib.rs
Comment (ahayzen-kdab) [inline crates/cxx-qt-build/src/lib.rs:367]: should we continue always adding `Core` or should this become explicit ? As you likely need to specify it on the CMake side?

Comment (LeonMatthesKDAB) [reply crates/cxx-qt-build/src/lib.rs:367]: This should be explicit.
Basically, move this `insert` into the `else` branch of the `if let` below.
If we're getting stuff from CMake, make sure it's exactly as specified by CMake.

File: examples/demo_threading/CMakeLists.txt
Comment (LeonMatthesKDAB) [inline examples/demo_threading/CMakeLists.txt:49]: We can't merge this of course, so make sure we undo that before merging :)

File: examples/demo_threading/CMakeLists.txt
Comment (LeonMatthesKDAB) [inline examples/demo_threading/CMakeLists.txt:56]: Can we now maybe remove some of the dependencies from the actual app?
E.g. QuickControls2 probably is only needed by our own QML code, right?
Not by main.cpp

File: crates/cxx-qt-build/src/lib.rs
Comment (LeonMatthesKDAB) [inline crates/cxx-qt-build/src/lib.rs:466]:
```suggestion
            panic!("Qt module mismatch between cxx-qt-build and CMake!\n\
                        Qt module '{module}' was not specified in CMake!\n\
                        When building with CMake, all Qt modules must be specified with the QT_MODULES argument in cxx_qt_import_crate");
```
I always find it worthwhile to make these errors as "foolproof" as possible :)
At the most basic level it will hopefully help understand our own design decisions a few months from now :sweat_smile:

**Review by LeonMatthesKDAB — COMMENTED** [review]
Mostly looks good, some minor nitpicks :)

File: examples/demo_threading/CMakeLists.txt
Comment (ahayzen-kdab) [reply examples/demo_threading/CMakeLists.txt:56]: In theory, although depends if we want to be rebuilding cxx-qt-lib for every example (as is happening now) or just enable qt_full + full everywhere. I'll see if i can match some of them.

File: examples/demo_threading/CMakeLists.txt
Comment (ahayzen-kdab) [reply examples/demo_threading/CMakeLists.txt]: done

File: crates/cxx-qt-build/src/lib.rs
Comment (ahayzen-kdab) [reply crates/cxx-qt-build/src/lib.rs]: done

**Review by LeonMatthesKDAB — APPROVED** [review]

**Pull request merged**
